cdf_fabric_replicator/time_series.py

```diff
@@ -241,17 +253,34 @@ def write_pd_to_deltalake(
     ) -> None:
         token = self.get_token()
 
-        write_deltalake(
-            table_or_uri=table,
-            data=df,
-            mode=mode,
-            engine="rust",
-            schema_mode="merge",
-            storage_options={"bearer_token": token, "use_fabric_endpoint": "true"},
-        )
-        return None
+        try:
+            write_deltalake(
+                table_or_uri=table,
+                data=df,
+                mode=mode,
+                engine="rust",
+                schema_mode="merge",
+                storage_options={"bearer_token": token, "use_fabric_endpoint": "true"},
+            )
+        except Exception as e:
+            logging.error(f"Error writing to Delta Lake: {e}")
+            raise e
 
     def get_token(self) -> str:
         return self.azure_credential.get_token(
             "https://storage.azure.com/.default"
         ).token
+
+    def create_subscription(
+        self, external_id: str, name: str, num_partitions: int
+    ) -> None:
+        logging.debug(f"Subscription {external_id} not found. Creating subscription...")
+
+        sub = DataPointSubscriptionWrite(
+            external_id=external_id,
+            name=name,
+            partition_count=num_partitions,
+            filter=flt.Exists(TimeSeriesProperty.external_id),
+        )
+        self.cognite_client.time_series.subscriptions.create(sub)
```

Comment:
Thinking about single responsibility, does it make sense to have all methods for creating subscription in the time series.py, might be worth some thought so see if we need a subscription.py

Reply:
Moved create_subscription to subscription.py